src/clib/grackle_types.h

```diff
@@ -19,11 +19,13 @@
 /
 ************************************************************************/
 
-#ifdef CONFIG_BFLOAT_4
+#include "grackle_float.h"
+
+#if defined CONFIG_BFLOAT_4 || defined GRACKLE_FLOAT_4
 #define gr_float float
 #endif
```

Comment:
I just wanted to make a small suggestion. Frankly, I'm not convinced that this change is worth making (since it's a little messy), but I figured that I would suggest it any way.

What if you changed this to read:
```c
#if defined CONFIG_BFLOAT_4 || defined GRACKLE_FLOAT_4
#define gr_float double

// simple error checking:
#if defined GRACKLE_FLOAT_8
#error "the CONFIG_BFLOAT_4 macro can't be defined when Grackle is compiled with 64bit floats"
#elif defined CONFIG_BFLOAT_8
#error "the CONFIG_BFLOAT_8 macro can't be defined when Grackle is compiled with 32bit floats"
#endif
#endif
```

Reply:
Yeah, I could do something like that for some error checking. I'll add something like that in.

I don't have strong feelings about deprecating CONFIG_BFLOAT. I suppose the reason to do it is that CONFIG_BLFOAT is rather nondescript and not obviously Grackle-related. I guess I can't think of too many instances where removing it altogether would create problems for users since it would just leave them with an extraneous definition and nothing else. Do you have a preference?

Reply:
Ideally, I think it would make sense to use `GRACKLE_FLOAT_[48]` everywhere instead of using both that macro and `CONFIG_BLFOAT_[48]` (If for no other reason than there is one less variable to keep track of). More pragmatically, doing that is probably not worth the effort for this PR.

It's totally up to you if you think the error-checking would be useful (I'm not totally convinced it provides enough utility to warrant inclusion -- after all, it looks pretty ugly)

Reply:
After looking through the code, I saw several places checking the definition of CONFIG_BFLOAT_[48] and decided we were better off getting rid of it entirely. I've added some simple checking for simultaneous definitions of GRACKLE_FLOAT_4 and GRACKLE_FLOAT_8 as well. I've tested this in a standalone executable and it seems good.